msv/src/main/java/com/sun/msv/grammar/SimpleNameClass.java

```diff
@@ -55,16 +56,23 @@ public SimpleNameClass( StringPair name ) {
     }
     
     public SimpleNameClass( String namespaceURI, String localName ) {
+        this.namespaceURI    = namespaceURI; 
+        this.localName        = localName;
+        this.prefix = null;
+    }
+        
+    public SimpleNameClass( String namespaceURI, String localName, String prefix ) {
         this.namespaceURI    = namespaceURI;
         this.localName        = localName;
+        this.prefix = prefix;
     }
     
     public StringPair toStringPair() {
         return new StringPair(namespaceURI,localName);
     }
     
     public String toString() {
-        if( namespaceURI.length()==0 )    return localName;
+        if( prefix != null )    return prefix + ":" + localName;
```

Comment:
but this function will turn a "" prefix string into ":localname" qName which seems not ideal? either change it to check that prefix isn't empty string or init the class with nulls

Reply:
Well spottet, I will keep the != null (being defensive) and will add an _&& !prefix.isEmpty()_